sdk/core/Azure.Core.Experimental/src/DynamicRequest.cs

```diff
@@ -0,0 +1,131 @@
+    {
+        private Request Request { get; }
+        private HttpPipeline HttpPipeline { get; }
+
+        private static readonly Encoding Utf8NoBom = new UTF8Encoding(false, true);
+
+        /// <inheritdoc />
+        public override RequestContent? Content
+        {
+            get => DynamicContent.Create(Body);
+
+            set {
+                MemoryStream ms = new MemoryStream();
+                if (value != null)
+                {
+                    value.WriteTo(ms, default);
+                    ms.Seek(0, SeekOrigin.Begin);
+                }
+                using (StreamReader sr = new StreamReader(ms, Utf8NoBom))
+                {
+                    Body = new DynamicJson(sr.ReadToEnd());
+                }
+                Request.Content = value;
+            }
+        }
+
+        // TODO(matell): How does the initialization here play into the ability to send a request with an "empty" body?
+        /// <summary>
+        /// The JSON body of request.
+        /// </summary>
+        public DynamicJson Body { get; set; } = DynamicJson.Object();
+
+        // TODO(matell): In Krzysztof's prototype we also took DiagnosticScope as a parameter, do we still need that?
+        /// <summary>
+        /// Creates an instance of <see cref="RequestContent"/> that wraps <see cref="DynamicJson"/> content.
+        /// </summary>
+        /// <param name="request">The <see cref="Request"/> to send.</param>
+        /// <param name="pipeline">The HTTP pipeline for sending and receiving REST requests and responses.</param>
+        public DynamicRequest(Request request, HttpPipeline pipeline)
+        {
+            Request = request;
+            HttpPipeline = pipeline;
+        }
+
+        /// <summary>
+        /// Send the request asynchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The response dynamically typed in a <see cref="DynamicResponse"/>.</returns>
+        public async Task<DynamicResponse> SendAsync(CancellationToken cancellationToken = default)
+        {
+            // Since we are sending the underlying request, we need to copy the Content on to it, or we'll lose the body.
+            Request.Content = Content;
+
+            Response res = await HttpPipeline.SendRequestAsync(Request, cancellationToken).ConfigureAwait(false);
+            DynamicJson dynamicContent;
+
+            if (res.ContentStream != null)
+            {
+                using (StreamReader sr = new StreamReader(res.ContentStream, encoding: Utf8NoBom, detectEncodingFromByteOrderMarks:true, bufferSize: 1024, leaveOpen: true))
+                {
+                    dynamicContent = new DynamicJson(await sr.ReadToEndAsync().ConfigureAwait(false));
+                }
+            }
+            else
+            {
+                //TODO(chamons) - Is this correct?
+                dynamicContent = new DynamicJson("");
```

Comment:
Why not `null`?

Reply:
FWIW, I agree.  It feels like when res.ContentStream is null, we should just have an null value for the "Body" part of the dynamic response,

Reply:
I wasn't sure if exposing a null to the user was acceptable here. Good to know.

Reply:
As I modified it, now I remember:

```
#nullable enable
...
    //
    // Summary:
    //     Represents a result of Azure operation.
    //
    // Type parameters:
    //   T:
    //     The type of returned value.
    [DebuggerTypeProxy (typeof (Azure.ResponseDebugView<>))]
    public abstract class Response<T>
    {
        protected Response ();

        //
        // Summary:
        //     Gets the value returned by the service.
        public abstract T Value { get; }
```
Value on Response is not marked as nullable, so if I mark:
```
        public override DynamicJson? Value { get; }
```
I get:
```
error CS8764: Nullability of return type doesn't match overridden member (possibly because of nullability attributes).
```

Reply:
So I can slap the type system around to for it, but according to our attributes Response.Value should not be null.

Reply:
Ah, the beauty of nullability annotations.